packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -249,6 +241,58 @@ export interface GraphQLApiProps {
 
 }
 
+/**
+ * A IamResource Class for IAM Permissions
+ */
+export class IamResource {
+  /**
+   * Generate the resourceNames given custom arns
+   *
+   * @param arns The custom arns that need to be permissioned
+   */
+  public static custom(...arns: string[]): IamResource {
```

Comment:
Does it make sense to call this without any `arns` passed, so `IamResource .custom()`?

I suspect it's not, so I can see two ways of handling that:

1. Add a non-variadic argument, like:

```ts
  public static custom(arn: string, ...arns: string[]): IamResource { ... }
```

2. Add a runtime validation that a 0 amount of parameters wasn't passed to `custom`.

I'll let you decide which one you like more 🙂.